### ISSUE TITLE
review logging for potential information leaks or excessive verbosity

### ISSUE THREAD

**Issue by daira** [zcash/zcash#232]

**Comment by bitcartel** [comment]
Restrict this ticket to logging that we've added.

**Comment by str4d** [comment]
Subticket: #1504

**Comment by str4d** [comment]
I'm reviewing this by running `git log -GLogPrint -p --full-diff v0.11.2..HEAD` and then searching for `LogPrint` in the resulting diffs.

**Comment by str4d** [comment]
I've scanned the output of the above command, and could not find any information leaks that were not protected by a debug flag (the only two I could see were `zrpc` and `zrpcunsafe`).

ACK

**Comment by daira** [comment]
I reviewed every use of ``LogPrint[f]`` under ``src/`` (excluding tests). I could not see any any logging of private information, in code we've changed or otherwise, subject to the following caveats:

* There may be deanonymizing information that fairly uniquely identifies the node doing the logging, but I did not count as "private information" for this review.
* I also did not count logging of amounts of ZEC in transactions/notes/error messages, or of which transactions are relevant to this wallet (because they involve an address for which the wallet has a key).
* In some cases exception messages were logged as ``e.what()``, and I could not review what was in the message because exceptions are non-local.
* The "bench" log category could potentially record information useful for timing attacks.
* The "zrpc" log category records z-addresses and private ZEC amounts.
* The "zrpcunsafe" log category also records portions of memo fields.
